include/swift/AST/Attr.def

```diff
@@ -508,6 +508,11 @@ DECL_ATTR(differentiable, Differentiable,
   ABIStableToAdd | ABIBreakingToRemove | APIStableToAdd | APIBreakingToRemove,
   91)
 
+DECL_ATTR(differentiating, Differentiating,
```

Comment:
Bikeshed language design issue - feel free to punt to the formal review process, but this attribute is textually very similar to @differentiable.  Did you consider something like `@derivativeOf` as the name of this attribute?

Reply:
I'm very open to this idea! Personally, I like `@derivative(of: ...)` and `@transpose(of: ...)` better than `@differentiating(...)` and `@transposing(...)`.

Reply:
+1 for `@derivative(of: ...)`

Reply:
`@derivative(of: ...)` does clearly evoke derivative registration! I'll rename to `@differentiating(...)` to `@derivative(of: ...)`, and deprecate `@differentiating(...)` on `tensorflow` branch with a warning, if there are no objections.

Reply:
+1

Reply:
Renamed `@differentiating` to `@derivative(of:)` in https://github.com/apple/swift/pull/28321/commits/4b9f025351afef770a431ae276581d5cf473e736.